This doesn't appear to work

Following the instructions I git cloned and attempted to npm install after cding into the directory.

Here are the errors I get.

4 warnings and 1 error generated.
make: *** [Release/obj.target/leveldown/src/batch.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
gyp ERR! System Darwin 16.7.0
gyp ERR! command "/usr/local/Cellar/node/10.0.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/patientplatypus/Documents/zennify.me/ipfsEXPERIMENTAL/ipfs-pubsub-room-demo/node_modules/leveldown
gyp ERR! node -v v10.0.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm WARN dirty-chai@2.0.1 requires a peer of chai@>=2.2.1 <5 but none is installed. You must install peer dependencies yourself.
npm WARN ipfs-pubsub-demo@0.0.1 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! leveldown@1.9.0 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the leveldown@1.9.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/patientplatypus/.npm/_logs/2018-08-31T16_59_58_866Z-debug.log

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
   "dependencies": {
     "browserify": "^14.4.0",
     "http-server": "^0.10.0",
-    "ipfs": "^0.27.3",
-    "ipfs-pubsub-room": "^1.1.3"
+    "ipfs": "^0.31.7",
+    "ipfs-pubsub-room": "^1.2.1"
   }
 }

